sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobContainerListDetails.java

```diff
@@ -42,13 +45,53 @@ public BlobContainerListDetails setRetrieveMetadata(boolean retrieveMetadata) {
         return this;
     }
 
+    /**
+     * Whether deleted containers should be returned.
+     *
+     * @return a flag indicating whether deleted containers should be returned
+     */
+    public boolean getRetrieveDeleted() {
+        return this.retrieveDeleted;
+    }
+
+    /**
+     * Whether deleted containers should be returned.
+     *
+     * @param retrieveDeleted Flag indicating whether deleted containers should be returned.
+     * @return the updated ContainerListDetails object
+     */
+    public BlobContainerListDetails setRetrieveDeleted(boolean retrieveDeleted) {
+        this.retrieveDeleted = retrieveDeleted;
+        return this;
+    }
+
     /**
      * @return the listing flags
+     * @deprecated use {@link #toIncludeTypes()}
      */
+    @Deprecated
```

Comment:
FYI - Deprecation needs to be called out and discussed in more detail whenever it occurs.

Reply:
I added more verbose description in `@deprecated` JavaDoc section. Is there any other place we should be calling this out?

Reply:
Generally - when review requests are made, explicitly mentioning deprecations is the best practice. Eventually I would imagine tooling will be added to our repo to make it automatic, but it's still a good idea to surface it ASAP so we don't make a decision without full consideration.